### PR TITLE
Add Realtime Checking Function

### PR DESCRIPTION
**Pull request by royeom**
New functions are below:

**Check username duplication**  : In  `/manager/data.py`, Check that the username entered as input already exists as a key in the DB. And send a result if or not.

**Check auth or not**  : In  `/restful/user/check/auth.py`, Check that which key exists in the session. Send a true or false of Github and LinkedIn.

Please take a look.

### CODE REVIEW

File: backend/quadcore/manager/data.py [hunk 1]
```diff
@@ -30,26 +30,40 @@ def set_user_profile(cls, info):
         """
         if info["username"] == None or info["email"] == None:
             return None
-        return cls.db.hmset("user:" + info["email"], info)
+        return cls.db.hmset("user:" + info["username"], {
+            'email': info["email"],
+            'fristName': info["firstName"],
```
Comment (harrydrippin) [inline backend/quadcore/manager/data.py]: `fristName` to `firstName`!

File: backend/quadcore/manager/data.py [hunk 2]
```diff
@@ -30,26 +30,40 @@ def set_user_profile(cls, info):
...
-    #     """
-    #     Check this email is already exist in database.
-    #     """
-    #     return cls.db.keys("user:" + email)
 
     @classmethod
-    def get_user_interests(cls, email):
+    def set_user_map(cls, info):
+        """
+        Connect username and email
```
Comment (harrydrippin) [inline backend/quadcore/manager/data.py]: Add period at the end of this sentence.

File: backend/quadcore/manager/data.py [hunk 3]
```diff
@@ -30,26 +30,40 @@ def set_user_profile(cls, info):
...
+        if info["username"] == None or info["email"] == None:
+            return None
+        return cls.db.hmset("email:" + info["email"], {
+            'username': info["username"] 
+        })
+    
+    @classmethod
+    def check_username(cls, username):
+        """
+        Check username in Real-time
```
Comment (harrydrippin) [inline backend/quadcore/manager/data.py]: Same as above.

**Review by harrydrippin — COMMENTED** [review]
Thanks! Please check my comments.

File: backend/quadcore/manager/data.py (hunk 1 above)
Comment (royeom) [reply backend/quadcore/manager/data.py]: I modified.. How funny !! :)

**Review by harrydrippin — APPROVED** [review]
LGTM

**Pull request merged**